### PR TITLE
RF_00.002.01 | New Item > Create Pipeline Project | Special character…

### PR DESCRIPTION
**Pull request by antnikol**
RF_00.002.01 | New Item > Create Pipeline Project | Special characters are not allowed in the project name
[#368](https://github.com/RedRoverSchool/JenkinsQA_JS_2024_fall/issues/368)

### CODE REVIEW

File: cypress/fixtures/pomFixtures/newJobPage.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+    "project_name_invalid": "New<>Name",
+    "new_item_name_invalid_message": "» ‘<’ is an unsafe character",
```
Comment (adiuzhikov) [inline cypress/fixtures/pomFixtures/newJobPage.json:3]: We have a file fixtures/messages.json for all types of messages, errors, and warnings. Please transfer this data to that file. 

Additionally, please adhere to the naming conventions for variables, objects, and arrays, which are camelCase, not snake_case.

File: cypress/pageObjects/NewJobPage.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+
+
+class NewJobPage
+{
+   getItemNameTextField = () => cy.get("#name.jenkins-input")
```
Comment (adiuzhikov) [inline cypress/pageObjects/NewJobPage.js:5]: I would rename this method to `getItemNameInputField`

File: cypress/e2e/newItemCreatePipelineProjectPOM.cy.js [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+describe("US_00.002 | New Item > Create Pipeline Project", () => {
+
+
+  let projectName = "New Pipeline"
+
+
+  it("TC_00.002.01 | Special characters are not allowed in the project name", () => {
+    dashboardPage
+      .clickNewItemMenuLink()
+      .typeNewItemName(project_name_invalid)
```
Comment (adiuzhikov) [inline cypress/e2e/newItemCreatePipelineProjectPOM.cy.js:30]: This method (`typeNewItemName()`) is located in the `NewJobPage.js`. Please, fix that

**Review by adiuzhikov — CHANGES_REQUESTED** [review]

File: cypress/fixtures/pomFixtures/newJobPage.json (hunk 1 above)
Comment (antnikol) [reply cypress/fixtures/pomFixtures/newJobPage.json]: Thanks for your attention. I transferred the error message to the messages.json, and renamed the other two variables in camelCase

File: cypress/e2e/newItemCreatePipelineProjectPOM.cy.js (hunk 3 above)
Comment (antnikol) [reply cypress/e2e/newItemCreatePipelineProjectPOM.cy.js]: When I sent my pull request, there was no other page NewJobPage.js

**Comment by adiuzhikov** [comment]
Please, resolve the conflicts so that we can see whether the test passes on CI.

File: cypress/pageObjects/NewJobPage.js [hunk 4]
```diff
@@ -4,24 +4,17 @@ class NewJobPage
...
         this.getFreeStlPrjType().click()
         return this
     }
-
-    okBtnClick()
-    {
-        this.getOKBtn().click()
-        return new ProjectConfigure()
-    }
-
```
Comment (adiuzhikov) [inline cypress/pageObjects/NewJobPage.js:25]: You have deleted okBtnClick(). Are you sure nobody is using it?

**Comment by antnikol** [comment]
> Please, resolve the conflicts so that we can see whether the test passes on CI.

Yes, you are right. My fault. I restored it method.

File: cypress/pageObjects/NewJobPage.js (hunk 4 above)
Comment (antnikol) [reply cypress/pageObjects/NewJobPage.js]: Yes, you are right. My fault. I restored it method.

**Review by adiuzhikov — APPROVED** [review]
Looks good

**Review by Sandarella — APPROVED** [review]

**Pull request merged**